CodingChallenges/CC_136_Polar_Noise_Loop_2/P5/sketch.js

```diff
@@ -0,0 +1,45 @@
+// Perlin Noise GIF Loops
+// Daniel Shiffman
+// https://thecodingtrain.com/CodingChallenges/136.2-perlin-noise-gif-loops.html
+// https://youtu.be/c6K-wJQ77yQ
+
+const totalFrames = 480;
+let counter = 0;
+let record = false;
+
+let particles = new Array(100);
+
+function setup() {
+  createCanvas(600, 600);
+  for (let i = 0; i < particles.length; i++) {
+    particles[i] = new Particle();
+  }
+
+
+}
+
+function draw() {
+  let percent = 0;
+  if (record) {
+    percent = float(counter) / totalFrames;
+  } else {
+    percent = float(counter % totalFrames) / totalFrames;
+  }
+  render(percent);
+  if (record) {
+    saveFrame("output/gif-" + nf(counter, 3) + ".png");
```

Comment:
I wonder if the p5 ports should not actually save any files but just run the animation? Or does this perform ok? I don't think `saveFrame()` is a p5 function btw.

Reply:
> I wonder if the p5 ports should not actually save any files but just run the animation?

Yes, that makes sense.